enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,377 @@
+   rather than using namespace VNIDs, since packets from ovn nodes will have
+   the VNID 0 set.
+2. To be compatible with Windows node VXLAN implementation, OVN-K hybrid overlay
+   uses the host interface MAC as VXLAN inner MAC. When packets arrive at the
+   br0 of the SDN node, they cannot be forwarded to the pod interface, due to
+   the MAC mismatch. We need to add flows for each pod which swaps the dst MAC
+   to the pod interface mac.
+
+### The Traffic Path
+
+#### Packets going from OpenShift SDN to OVN Kubernetes
+
+On the SDN side, it doesn't need to know if the peer node is a SDN node or a OVN
+node. We reuse the existing VXLAN tunnel rules one the SDN side.
+- Egress NetworkPolicy rules and service proxying happen as normal
+- When the packet reaches table 90, it will hit a "send via vxlan" rule that was
+  generated based on a HostSubnet object.
+
+On the OVN side:
+- OVN accepts the packet via the VXLAN tunnel, ignore the VNID set by SDN and
+  then just routes it normally.
+- Ingress NetworkPolicy processing will happen when the packet reaches the
+  destination pod's switch port, just like normal.
+  - Our NetworkPolicy rules are all based on IP addresses, not "logical input
+    port", etc, so it doesn't matter that the packets came from outside OVN and
+    have no useful OVN metadata.
+
+#### Packets going from OVN Kubernetes to OpenShift SDN
+
+On the ovn side:
+- The packet just follow the same path of hybrid overlay. The packet just has to
+  get routed out the VXLAN tunnel with VNID 0.
+
+On the sdn side:
+- We have to change ingress NetworkPolicy processing to be based entirely on pod
+  IPs rather than using namespace VNIDs, since packets from ovn nodes won"t have
+  the VNID set. There is already code to generate the rules that way though,
+  because egress NetworkPolicy already works that way.
+
+### Workflow Description
+
+#### Migration Setup
+
+1. The admin kicks off the migration process by updating the custom resource
+   `network.operator`:
+   
+   ```bash
+   $ oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration":{"networkType":"OVNKubernetes","isLive":true}}}"
+   ```
+   
+   CNO will check if there is any unsupported feature (refer the
+   [Limitations](#limitations) section) enabled in the cluster. If yes, it will
+   set an warning message under the `status.migration` field of the custom
+   resource `network.operator`.
+
+2. CNO will redeploy the openshift-sdn in the __migration mode__. It will add a
+   condition check to the wrapper script of the sdn DaemonSet, to see if the
+   bridge `br-ex` exists on the node . If `br-ex` exists, it means the node has
+   already been updated by MCO, therefore it's ready for running OVN Kubernetes,
+   the sdn pod would just do "sleep infinity" rather than actually launching the
+   `openshift-sdn-node` process.
+
+3. CNO will also deploy OVN Kubernetes to the cluster with hybrid overlay
```

Comment:
so at this step will ovnkube-master be brought up on the control nodes without blocking/waiting for anything?

Reply:
No, it doesn't need to. At this moment, `ovnkube-node` process won't be running on any node. We only have the ovn-k control-plane deployed.